Add `partiql-lang-kotlin` major version migration guides (#688)
partiql/partiql-lang-kotlin: docs/dev/version-migration/v0.6-to-v0.7-migration.md

# v0.7.* (latest v0.7.0)

## New features

> * 2.3x more performance `LIKE` operator

Reviewer: ```suggestion
* 2.3x more performant `LIKE` operator
```

* An experimental query planner API along with logical and physical plans structures with the support of non-default physical operator implementations.

> * Enable `MIN` and `MAX` to work with all the data-types.

Reviewer: ```suggestion
* `MIN` and `MAX` now work with all data types.
```
